src/Parser/GeoJsonParser.js

```diff
@@ -42,54 +42,54 @@ const toFeature = {
         }
         geometry.updateExtent();
     },
-    default(feature, crsIn, coordsIn, filteringExtent, setAltitude, properties) {
+    default(feature, crsIn, coordsIn, filteringExtent, setAltitude, properties, parentStyle) {
         if (filteringExtent && firstPtIsOut(filteringExtent, coordsIn, crsIn)) {
             return;
         }
 
         const geometry = feature.bindNewGeometry();
         geometry.properties = properties;
-        geometry.properties.style = new Style().setFromGeojsonProperties(properties, feature.type);
+        geometry.properties.style = new Style({}, parentStyle).setFromGeojsonProperties(properties, feature.type);
         this.populateGeometry(crsIn, coordsIn, geometry, setAltitude, feature);
         feature.updateExtent(geometry);
     },
-    polygon(feature, crsIn, coordsIn, filteringExtent, setAltitude, properties) {
+    polygon(feature, crsIn, coordsIn, filteringExtent, setAltitude, properties, parentStyle) {
         // filtering
         if (filteringExtent && firstPtIsOut(filteringExtent, coordsIn[0], crsIn)) {
             return;
         }
         const geometry = feature.bindNewGeometry();
         geometry.properties = properties;
-        geometry.properties.style = new Style().setFromGeojsonProperties(properties, feature.type);
+        geometry.properties.style = new Style({}, parentStyle).setFromGeojsonProperties(properties, feature.type);
 
         // Then read contour and holes
         for (let i = 0; i < coordsIn.length; i++) {
             this.populateGeometry(crsIn, coordsIn[i], geometry, setAltitude, feature);
         }
         feature.updateExtent(geometry);
     },
-    multi(type, feature, crsIn, coordsIn, filteringExtent, setAltitude, properties) {
+    multi(type, feature, crsIn, coordsIn, filteringExtent, setAltitude, properties, parentStyle) {
```

Comment:
could you pass parent style when you instance feature?

Reply:
Done